File: urbanpulse-frontend/src/services/complaintService.js
import axiosInstance from "../api/axiosConfig";

const complaintService = {

  getAllComplaints: async () => {
    const response = await axiosInstance.get("/complaints");
    return response.data;
  },

  createComplaint: async (complaint) => {
    const response = await axiosInstance.post(
      "/complaints",
      complaint
    );
    return response.data;
  },

  updateComplaint: async (id, complaint) => {
    const response = await axiosInstance.put(
      `/complaints/${id}`,
      complaint
    );
    return response.data;
  },

  deleteComplaint: async (id) => {
    const response = await axiosInstance.delete(
      `/complaints/${id}`
    );
    return response.data;
  }

};

export default complaintService;

File: urbanpulse-backend/src/main/java/com/urbanpulse/service/ComplaintService.java
package com.urbanpulse.service;

import com.urbanpulse.entity.Complaint;
import java.util.List;

public interface ComplaintService {

    Complaint saveComplaint(Complaint complaint);

    List<Complaint> getAllComplaints();

    Complaint getComplaintById(Long id);

    Complaint updateComplaint(Long id, Complaint complaint);

    void deleteComplaint(Long id);
}